scripts/cluster/start.sh

```diff
@@ -82,3 +82,36 @@ kubectl apply -f "${DIR}/config/daemon.yaml"
 kubectl apply -f "${DIR}/config/network-policy.yaml"
 kubectl apply -f "${DIR}/config/allow-dns.yaml"
 kubectl patch ServiceAccount default --patch "automountServiceAccountToken: false"
+
+# Cloud DNS
+
+if [ ! -z "${DOMAIN_NAME}" ]
+then
+  DNS_ZONE=$(gcloud dns managed-zones list --filter "dns_name:${DOMAIN_NAME}" --format 'get(name)')
+  if [ -z "${DNS_ZONE}" ]; then
+    gcloud dns managed-zones create "${CLUSTER_NAME}-dns-zone" --description "DNS Zone for ${DOMAIN_NAME}" --dns-name="${DOMAIN_NAME}."
+  fi
+
+  GSA_NAME="kctf-cloud-dns"
```

Comment:
could you change the name of the env var? maybe DNS_GSA_NAME? (and same for GCS_GSA_NAME)? so it's clear they are gonna be different